astropy/io/registry.py

```diff
@@ -599,3 +604,106 @@ def _get_valid_format(mode, cls, path, fileobj, args, kwargs):
                 ', '.join(sorted(valid_formats, key=itemgetter(0)))))
 
     return valid_formats[0]
+
+
+class UnifiedReadWrite:
+    """Base class for the worker object used in unified read() or write() methods.
+
+    This lightweight object is created for each `read()` or `write()` call
+    via ``read`` / ``write`` descriptors on the data object class.  The key
+    driver is to allow complete format-specific documentation of available
+    method options via a ``help()`` method, e.g. ``Table.read.help('fits')``.
+
+    Subclasses must define a ``__call__`` method which is what actually gets
+    called when the data object ``read()`` or ``write()`` method is called.
+
+    For the canonical example see the `~astropy.table.Table` class
+    implementation (in particular the ``connect.py`` module there).
+
+    Parameters
+    ----------
+    instance : object
+        Descriptor calling instance or None if no instance
+    cls : type
+        Descriptor calling class (either owner class or instance class)
+    method_name : str
+        Method name, either 'read' or 'write'
+    """
+    def __init__(self, instance, cls, method_name):
+        self._instance = instance
+        self._cls = cls
+        self._method_name = method_name  # 'read' or 'write'
+
+    def help(self, format=None, out=None):
+        """Output help documentation for the specified unified I/O ``format``.
+
+        By default the help output is printed to the console via ``pydoc.pager``.
+        Instead one can supplied a file handle object as ``out`` and the output
+        will be written to that handle.
+
+        Parameters
+        ----------
+        format : str
+            Unified I/O format name, e.g. 'fits' or 'ascii.ecsv'
+        out : None or file handle object
+            Output destination (default is stdout via a pager)
+        """
+        cls = self._cls
+        method_name = self._method_name
+
+        # Get reader or writer function
+        get_func = get_reader if method_name == 'read' else get_writer
+        try:
+            if format:
+                read_write_func = get_func(format, cls)
+        except IORegistryError as err:
+            reader_doc = 'ERROR: ' + str(err)
+        else:
+            # General docs
+            header = ('{}.{} general documentation\n'
+                      .format(cls.__name__, method_name))
+            reader_doc = re.sub('.', '=', header)
+            reader_doc += header
+            reader_doc += re.sub('.', '=', header)
```

Comment:
Can you add a newline after the heading? That is,

```
================================
Table.read general documentation
================================

Read and parse a data table and return as a Table.
```

instead of

```
================================
Table.read general documentation
================================
Read and parse a data table and return as a Table.
```

Reply:
Done.